packages/adaptive_scaffold/lib/src/adaptive_layout.dart

```diff
@@ -0,0 +1,517 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui';
+import 'package:flutter/widgets.dart';
+import 'breakpoints.dart';
+import 'slot_layout.dart';
+
+enum _SlotIds {
+  primaryNavigation,
+  secondaryNavigation,
+  topNavigation,
+  bottomNavigation,
+  body,
+  secondaryBody,
+}
+
+/// [AdaptiveLayout] is dedicated to simplifying the process of developing
```

Comment:
I would hope that the first paragraph below the introductory sentence would explain what kinds of apps is this component intended for? Is it suitable for all applications? Seems like it's intended for applications with certain limits that follow the Material guidelines.